libPhoneNumber/NBPhoneNumber.m

```diff
@@ -48,8 +48,12 @@ - (BOOL)isEqualToObject:(NBPhoneNumber*)otherObj
 
 - (NSUInteger)hash
 {
-    NSData *selfObject = [NSKeyedArchiver archivedDataWithRootObject:self];
-    return [selfObject hash];
+    NSUInteger hash = 0;
+    hash ^= self.countryCode.hash + 0x9e3779b9 + (hash << 6) + (hash >> 2);
```

Comment:
I initialized `hash` to `self.countryCode.hash`. It didn't really need the `0x9e3779b9`. I also added the URL in a comment.